### ISSUE TITLE
:seq-limit for vectors?

### ISSUE THREAD

**Issue by niclasnilsson** [greglook/puget#50]
I really like this library and use it to color code REPL output. However, I have one problem that I'd be interested to hear the reasoning behind. 

:seq-limit only works for seqs (obvious from the name), but that means that vectors, maps and other things are still fully printed, even if they are huge. If I set *print-length* in a standard clojure REPL, if works for all sorts of collections (see below). Is there a specific reason that puget doesn't have a similar thing, or that :seq-limit could be applied to other collections (or to have be :coll-limit or similar if needed)? Is it something you'd be interested in a pull request for, and if so, do you have any preference of how to solve it?

```
user=> (set! *print-length* 3)
3
user=> (range 10)
(0 1 2 ...)
user=> [1 2 3 4 5 6 7 8 9]
[1 2 3 ...]
user=> {:a 1 :b 2 :c 3 :d 4 :e 5}
{:a 1, :b 2, :c 3, ...}```

**Comment by greglook** [comment]
Puget's `:seq-limit` setting was implemented to avoid realizing infinitely large collections, of which sequences are the only type. Since the library was (initially) more focused on _canonical rendering_ rather than pretty-printing, there wasn't a use-case for partially rendering large (but finite) vectors, maps, and sets.

That said, I agree with you that this would be a valuable addition to the library. Implementing it as `:coll-limit` for backwards-compatibility seems like a good idea, and sequences can use that setting as a fallback if `:seq-limit` is not set. This also has implications on the canonical sorting logic - obviously if you have a collection whose size is larger than `:coll-limit` you shouldn't try to sort the keys before printing it!

Absolutely open to a PR if you feel like tackling this. :+1:

(Thanks for your patience, I've been AFK travelling for a few weeks.)

**Comment by greglook** [comment]
Released in [1.3.0](https://github.com/greglook/puget/releases/tag/1.3.0).
